src/interfaces/coral_web/src/components/Agents/AgentForm.tsx

```diff
@@ -0,0 +1,136 @@
+  errors?: Partial<Record<AgentFormFieldKeys, string>>;
+  className?: string;
+};
+/**
+ * @description Base form for creating/updating an agent.
+ */
+export const AgentForm: React.FC<Props> = ({
+  fields,
+  onChange,
+  onToolToggle,
+  errors,
+  className,
+}) => {
+  const { data: deployments } = useListAllDeployments();
+  const { models } = useModels(fields.deployment);
+  const deploymentOptions: DropdownOptionGroups = [
+    {
+      options: (deployments ?? []).map(({ name }) => ({
+        label: name,
+        value: name,
+      })),
+    },
+  ];
+  const modelOptions: DropdownOptionGroups = [
+    {
+      options: models.map((model) => ({
+        label: model,
+        value: model,
+      })),
+    },
+  ];
+  const { data: toolsData } = useListTools();
+  const tools = toolsData?.filter((t) => t.is_available) ?? [];
+
+  useEffect(() => {
+    // need to wait for deployments to finish changing
+    // otherwise, there will be a race condition when
+    // setting the model
+    if (fields.deployment === DEPLOYMENT_COHERE_PLATFORM) {
+      onChange('model', MODEL_COMMMAND_R_PLUS);
+    } else {
+      onChange('model', '');
+    }
+  }, [fields.deployment]);
+
+  return (
+    <div className={cn('flex flex-col gap-y-4', className)}>
+      <InputLabel label="name" className="pb-2">
+        <Input
+          kind="default"
+          value={fields.name ?? ''}
+          placeholder="Give your assistant a name"
+          onChange={(e) => onChange('name', e.target.value)}
+          hasError={!!errors?.name}
+          errorText={errors?.name}
+        />
+      </InputLabel>
+      <InputLabel label="description" className="pb-2">
+        <Input
+          kind="default"
+          value={fields.description ?? ''}
+          placeholder="What does your assistant do?"
+          onChange={(e) => onChange('description', e.target.value)}
+        />
+      </InputLabel>
+      <InputLabel label="Preamble">
+        <textarea
+          value={fields.preamble ?? ''}
+          placeholder="Give instructions to your chatbot. What does it do? How does it behave?"
+          className={cn(
+            'mt-2 w-full flex-1 resize-none p-3',
+            'transition ease-in-out',
+            'rounded-lg border',
+            'bg-marble-100',
+            'border-marble-500 placeholder:text-volcanic-700 focus:border-secondary-700',
+            'focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-4 focus-visible:outline-volcanic-900'
+          )}
+          rows={5}
+          onChange={(e) => onChange('preamble', e.target.value)}
+          data-testid="input-preamble"
+        />
+      </InputLabel>
+      <Dropdown
+        className="w-full"
+        label="Deployment"
+        kind="default"
+        value={fields.deployment}
+        onChange={(deployment: string) => onChange('deployment', deployment)}
+        optionGroups={deploymentOptions}
+      />
+      {modelOptions[0].options.length > 0 && (
```

Comment:
not in designs but felt like it could be confusing to the user when there were no models available